src/components/App/index.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import './styles.css';
+
+import Display from '../Display';
+import ButtonPanel from '../ButtonPanel';
+
+import calculate from '../../services/calculate';
+
+export default class App extends React.Component {
+  constructor(props) {
```

Comment:
There's also a cleaner and shorter way to write our `class`:

```javascript
class App extends React.Component {
  state = {
     total: null,
     next: null,
     operation: null,
     calculated: false,
   }
}

handleClick = (buttonName) => {// same logic }

render () {
  // ditto
}
```

Reply:
I read some article codes written like this. But I wanted to go deeper and figure out how it works before using it. Here are my questions: How does your version handle handleClick without binding? (I also don't know why we using it, I'm gonna research why we bind the click methods, I can understand it's related to moving scope but I need to see why), what if I need to use props, should I change it when I need to use props?

Reply:
Changed it through your suggestion. Check the related [commit](https://github.com/btuerker/react-calculator/pull/1/commits/81f0e5334d4f9749efe03cf722740206e7854b94)